src/plugins/intel_cpu/src/emitters/tpp/x64/jit_tpp_emitter.cpp

```diff
@@ -0,0 +1,145 @@
+// Copyright (C) 2020-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "jit_tpp_emitter.hpp"
+
+#include "snippets/lowered/port_connector.hpp"
+#include "snippets/lowered/port_descriptor.hpp"
+#include "transformations/snippets/tpp/op/eltwise.hpp"
+
+using namespace Xbyak;
+using namespace dnnl::impl;
+using namespace dnnl::impl::cpu::x64;
+
+namespace ov {
+namespace intel_cpu {
+
+using jit_generator = dnnl::impl::cpu::x64::jit_generator;
+using cpu_isa_t = dnnl::impl::cpu::x64::cpu_isa_t;
+using ExpressionPtr = ov::snippets::lowered::ExpressionPtr;
+using ExpressionPort = snippets::lowered::ExpressionPort;
+
+VectorDims TppEmitter::get_projected_subtensor(const snippets::lowered::PortDescriptorPtr& desc) {
```

Comment:
Ideally yes, but subtensors are not used anywhere else. So it would be highly inefficient to write a separate transformation to update a couple port descriptors: we will spend most of the time on redundant `dynamic_cast`.

It's a valid point, but we need to address it when we will deprecate counts in favor of subtensors.